src/main/java/org/terasology/commonworld/Sector.java

```diff
@@ -35,27 +34,28 @@
     public static final int SIZE_X = 1024;
     public static final int SIZE_Z = 1024;
 
-    private final ImmutableVector2i coords;
+    private final Vector2i coords = new Vector2i();
 
-    private final Rect2i bounds;
+    private final BlockArea bounds = new BlockArea(BlockArea.INVALID);
 
 
     /**
      * @param coords the coordinates
      */
-    Sector(BaseVector2i coords) {
+    Sector(Vector2ic coords) {
         if (coords == null) {
             throw new NullPointerException("coords cannot be null");
         }
 
-        this.coords = ImmutableVector2i.createOrUse(coords);
-        this.bounds = Rect2i.createFromMinAndSize(coords.getX() * SIZE_X, coords.getY() * SIZE_Z, SIZE_X, SIZE_Z);
+        this.coords.set(coords);
+        this.bounds.set(0,0,0,0);
```

Comment:
Why is this needed? Isn't the line below enough to (1) set the block area to a specific position (with the old size), and (2) set the size from the minimum corner?